File: scripts/regenerate-lock.py
#!/usr/bin/env python3
import subprocess
import os

os.chdir('/vercel/share/v0-project')

# Run npm install to regenerate the lock file
print("Regenerating package-lock.json with npm install...")
result = subprocess.run(['npm', 'install'], capture_output=True, text=True)
print(result.stdout)
if result.stderr:
    print("STDERR:", result.stderr)
print(f"Return code: {result.returncode}")

File: scripts/regenerate-lock.mjs
import { execSync } from 'child_process';
import { unlinkSync, existsSync } from 'fs';
import { resolve } from 'path';

const projectDir = '/vercel/share/v0-project';
const lockFilePath = resolve(projectDir, 'package-lock.json');

try {
  console.log('[v0] Regenerating package-lock.json...');
  
  // Remove old lock file if it exists
  if (existsSync(lockFilePath)) {
    unlinkSync(lockFilePath);
    console.log('[v0] Removed old lock file');
  }

  // Run npm install to create a fresh lock file
  console.log('[v0] Running npm install...');
  execSync('npm install', { 
    cwd: projectDir,
    stdio: 'inherit'
  });

  console.log('[v0] Lock file regenerated successfully!');
} catch (error) {
  console.error('[v0] Error regenerating lock file:', error.message);
  process.exit(1);
}
